soufi/finders/rhel.py

```diff
@@ -0,0 +1,58 @@
+# Copyright (c) 2021 Cisco Systems, Inc. and its affiliates
+# All rights reserved.
+
+from typing import Iterable
+
+import soufi.finders.yum as yum_finder
+from soufi import finder
+
+DEFAULT_REPO = "https://cdn-ubi.redhat.com/content/public/ubi/dist/"
+
+
+class RHELFinder(yum_finder.YumFinder):
+    """Find Red Hat Enterprise Linux source files.
+
+    By default, uses the public UBI index at https://cdn-ubi.redhat.com/
+    """
+
+    distro = finder.Distro.rhel.value
+
+    # UBI is kind of sprawling, and they are fairly idiosyncratic with where
+    # they like to put their packages, tree-wise.  Once you find the proper
+    # subtree, they are delightfully regular with the repo structure, however.
+    # Rather than aimlessly poke around their CDN, we'll use this
+    # hand-curated list of search paths.
+    default_search_dirs = (
+        'ubi8/8/x86_64/baseos/',
+        'ubi8/8/x86_64/appstream/',
+        'ubi8/8/x86_64/codeready-builder/',
+        'ubi/server/7/7Server/x86_64/',
+        'ubi/server/7/7Server/x86_64/extras/',
+        'ubi/server/7/7Server/x86_64/devtools/1/',
+        'ubi/server/7/7Server/x86_64/optional/',
+        'ubi/server/7/7Server/x86_64/rhscl/1/',
+        'ubi/atomic/7/7Server/x86_64/',
+    )
```

Comment:
I figure of anything in this PR, this particular decision to "punt" source lookups for UBI has cause to be one of the more contentious, so I'll attempt to get in front of that be elaborating on what I have in the comments here:

As one can see from the provided list, Red Hat is fond of completely restructuring their repository layouts between major releases.  There is effectively zero commonality between UBI 7 and UBI 8, so it stands to reason that we will have absolutely no idea how to look for repos in UBI 9 when it ships.  This would infer that:

1. To enable built-in support for UBI 9 in the future, a new release of SouFi will be required regardless
2. To facilitate support for UBI 7+8+9 with this code, one would simply need to copy+paste the default search list, and the UBI 9 repo dirs, and initialize the finder with the whole list.

Given the above, I think the "bare minimum effort" approach taken here is justified.  It's also really, really fast. 😆